tests/test_dict.py

```diff
@@ -60,6 +64,69 @@ def test_gpt2_bpe_tokenize(self):
             u'Hello, ParlAI! \U0001f600',
         )
 
+    def test_byte_level_bpe_tokenize(self):
+        # test loading
+        download(DEFAULT_BYTELEVEL_BPE_VOCAB, './', 'test-byte-level-bpe-vocab.json')
+        download(DEFAULT_BYTELEVEL_BPE_MERGE, './', 'test-byte-level-bpe-merges.txt')
+        opt = Opt(
+            {
+                'dict_tokenizer': 'bytelevelbpe',
+                'datapath': './',
+                'bpe_vocab': './test-byte-level-bpe-vocab.json',
+                'bpe_merge': './test-byte-level-bpe-merges.txt',
+            }
+        )
+        agent = DictionaryAgent(opt)
+        self.assertEqual(
+            # grinning face emoji
+            agent.bytelevelbpe_tokenize(u'Hello, ParlAI! \U0001f600'),
+            [
+                'H',
+                'ello',
+                ',',
+                'Ġ',
+                'P',
+                'ar',
+                'l',
+                'A',
+                'I',
+                '!',
+                'Ġ',
+                'ð',
+                'Ł',
+                'ĺ',
+                'Ģ',
+            ],
+        )
+        self.assertEqual(
+            agent.vec2txt(
+                [
+                    agent.tok2ind[w]
+                    for w in [
+                        'H',
+                        'ello',
+                        ',',
+                        'Ġ',
```

Comment:
Oy - how does the G with the dot equal a space between "Hello," and "ParlAI!" below?

Reply:
It splits by byte level and then just cast it to Unicode I think.